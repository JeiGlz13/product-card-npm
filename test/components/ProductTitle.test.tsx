import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/product';


describe('Pruebas en el ProductTitle', () => { 
    test('debe mostrar el titulo personalizado', () => { 
        const wrapper = renderer.create(
            <ProductTitle title='Custom Product' desc='Product description' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
     });

     test('debe mostar el componente con el title del context', () => { 
        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    ()=>(
                        <ProductTitle/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
      });
 });
