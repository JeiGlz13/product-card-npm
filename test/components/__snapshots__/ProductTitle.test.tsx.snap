// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el ProductTitle debe mostar el componente con el title del context 1`] = `
<div
  className="mt-8 w-96 mx-2 md:p-4 p-2 bg-white rounded-3xl undefined"
>
  <div
    className="lg:w-5/12  flex flex-col justify-center mt-2 pb-8 lg:pb-0"
  >
    <h1
      className="text-gray-800 text-lg lg:text-2xl font-semibold undefined"
    >
      Coffee Without Image
    </h1>
    <p
      className="text-base leading-normal text-gray-600  mt-2 undefined"
    >
      Coffee is darkly colored, bitter, slightly acidic and has a stimulating effect in humans
    </p>
  </div>
</div>
`;

exports[`Pruebas en el ProductTitle debe mostrar el titulo personalizado 1`] = `
<div
  className="lg:w-5/12  flex flex-col justify-center mt-2 pb-8 lg:pb-0"
>
  <h1
    className="text-gray-800 text-lg lg:text-2xl font-semibold undefined"
  >
    Custom Product
  </h1>
  <p
    className="text-base leading-normal text-gray-600  mt-2 undefined"
  >
    Product description
  </p>
</div>
`;
